Packs/JARM/Integrations/JARM/JARM.py

```diff
@@ -0,0 +1,159 @@
+requests.packages.urllib3.disable_warnings()  # pylint: disable=no-member
+
+DEFAULT_PORT = 443
+
+""" CLIENT CLASS """
+
+
+class Client:
+    def jarm_fingerprint(self, host: str, port: int) -> Tuple[str, str, int]:
+        return Scanner.scan(host, port)
+
+
+""" HELPER FUNCTIONS """
+
+
+def parse_hostname(hostname: str, port: Optional[int]) -> Dict[str, Any]:
+    """
+    Parses a target hostname. Supports multiple ipv4/fqdn with and without port formats.
+    """
+    target: Dict[str, Any] = {}
+    if not hostname.startswith('https://'):
+        hostname = 'https://' + hostname
+
+    parsed_url = urlparse(hostname)
+    if port:
+        target['port'] = port
+    elif parsed_url.port:
+        target['port'] = parsed_url.port
+    else:
+        target['port'] = DEFAULT_PORT
+
+    try:
+        ip = ip_address(parsed_url.hostname)
+        target['target_host'] = str(ip)
+        target['target_type'] = 'ip'
+    except ValueError:
+        target['target_host'] = parsed_url.hostname
+        target['target_type'] = 'fqdn'
+
+    return target
+
+
+""" COMMAND FUNCTIONS """
+
+
+def test_module(client: Client) -> str:
+    return "ok"
+
+
+def jarm_fingerprint_command(
+    client: Client, args: Dict[str, Any]
+) -> List[CommandResults]:
+    class JARMDBotScore(Common.Indicator):
+        def __init__(self, output: Dict[str, Any]):
+            self._jarm = output.get('Fingerprint')
+            self._ip = output.get('IP')
+            self._port = output.get('Port')
+            self._fqdn = output.get('FQDN')
+
+        def to_context(self) -> Dict[str, Any]:
+            return {
+                "DBotScore": {
+                    "Indicator": self._jarm,
+                    "Type": "jarm",
+                    "Vendor": "JARM",
+                    "Score": Common.DBotScore.NONE,
+                    "Host": {
+                        'ip': self._ip,
+                        'fqdn': self._fqdn,
+                        'port': self._port
+                    }
+                }
+            }
+
+    host = args.get("host")
+    if not host:
+        raise ValueError("Host name (IP or domain) not specified")
+
+    port = arg_to_number(args.get("port"))
+
+    target = parse_hostname(host, port)
+
+    target_type = target.get('target_type')
+    if not target_type:
+        raise ValueError('Cannot determine scan target')
+
+    target_host = target.get('target_host')
+    if not target_host:
+        raise ValueError('Cannot determine scan target')
+
+    port = target.get('port')
+    if not port:
+        raise ValueError('Invalid port provided')
+
+    result = client.jarm_fingerprint(target_host, port)
+
+    if target_type == 'ip':
+        output = {"IP": target_host, "Port": port, "Fingerprint": result[0]}
+    elif target_type == 'fqdn':
+        output = {"FQDN": target_host, "Port": port, "Fingerprint": result[0]}
+    else:
+        raise RuntimeError('Invalid target type')  # this code should be unreachable
```

Comment:
I suggest removing this code or modifying it.
If this error is reached, it will not make sense to the user nor to a developer unfamiliar with the API.
If we deem this case as impossible - we should not address it IMO.

Reply:
done